### PR TITLE
fix: SPL stats on records for segments that do not have matched RRC

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- Fixes the issue where the records are not being sent to the Client, as the function that handles the Stats Aggregation keeps waiting for all the Segments to be processed.
- Also fixes the issue with `fillNull` command if a query had multiple `ColumnsRequest`.
- Refactored `fillNull` command implementation.

# Testing
- Tested by running the queries through the UI across multiple segments

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/search/searchaggs.go [hunk 1]
```diff
@@ -486,7 +486,7 @@ func PerformMeasureAggsOnRecs(nodeResult *structs.NodeResult, recs map[string]ma
 				continue
 			}
 
-			if !dtypeVal.IsNumeric() {
+			if !dtypeVal.IsNumeric() && mOp.MeasureFunc != utils.Count {
```
[the remark this answers is not in the document]
Comment (Macbeth98) [reply pkg/segment/search/searchaggs.go:489]: For count, the `field` value do not need to be numeric.

File: pkg/segment/reader/record/rrcreader.go [hunk 2]
```diff
@@ -35,31 +34,15 @@ import (
...
-	mapMutex.Lock()
-	defer mapMutex.Unlock()
-
-	// Check if the nodeRes instance exists for the given qid
-	if nr, exists := nodeResMap[qid]; exists {
-		return nr
+	nodeRes, err := query.GetOrCreateQuerySearchNodeResult(qid)
+	if err != nil {
+		nodeRes = &structs.NodeResult{}
 	}
```
[the remark this answers is not in the document]
Comment (Macbeth98) [reply pkg/segment/reader/record/rrcreader.go:41]: If there is an error, it means that the query is not `async`. Like how we get the `numTotalSegments`, this is also the same.

[the remark this answers is not in the document]
Comment (AndrewHess) [reply pkg/segment/reader/record/rrcreader.go:41]: Ok. Then can you add a comment to the code to note that

[the remark this answers is not in the document]
Comment (Macbeth98) [reply pkg/segment/reader/record/rrcreader.go]: I have added the comment, similar to what we have at `numTotalSegments`.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**